Lib/glyphsLib/classes.py

```diff
@@ -1602,7 +1602,7 @@ def metricsSource(self):
 
         # No custom parameters apply, go home
         if not source_master_id:
-            return self
+            return None
```

Comment:
the docstring above says "Normally this is the current master itself", hence this would `return self`, but now you changed it to None. `glyphsLib.classes` module is supposed to mimic the Glyphs.app's own Python Scripting API. Is there a corresponding `metricsSource` property in the latter, what is it supposed to return in this case? We should try to match that. I see you're using the `None` return value as a signal that there is no linked metrics master, maybe you could test `master.metricsSource is master`?

Reply:
There is no `GSMaster.metricsSource` property in the Glyphs API, it was added as a helper in #641 by @simoncozens.

Reply:
> I even just noticed something else: if source_master_id in self.font.masters: is comparing IDs against master objects, so that's nonsense.

Good catch, @yanone … I think we need to check all the master ids, like we already do for the names:

```python
    @property
    def metricsSource(self):
        # [...]

        # No custom parameters apply, go home
        if not source_master_id:
            return None

        # Try by master id
        for source_master in self.font.masters:
            if source_master.id == source_master_id:
                return source_master

        # Try by name
        for source_master in self.font.masters:
            if source_master.name == source_master_id:
                return source_master
```

Reply:
No, we can do it easier:

```python
        # Try by master id
        source_master = self.font.masterForId(source_master_id)
        if source_master is not None:
            return source_master

        # Try by name ...
```

Reply:
I guess this is confusing because the custom parameter doesn’t return the master ID, but the master name … so is the check by master ID needed at all?

Reply:
Regarding @anthrotype’s inquiry about the return value: The `.metricsSource` attribute is only ever used once, here:https://github.com/googlefonts/glyphsLib/blob/1903622df5d31d056f4a7a2186d27a4d643f62bb/Lib/glyphsLib/builder/glyph.py#L257-L259

Since the value is checked for `None`, I conclude that returning `None` is the correct answer.

Otherwise I’ve added a test that checks the width of brace and bracket layers.
Sadly, nowadays, the Actions-based tests don't complete because of Codecov. I'm guessing it's not sending the coverage because this is a PR, which makes sense. So 2 out of 4 test are cancelled subsequently.

The test itself passes, though. Please have a look at the test routines and decide how to move forward.

Reply:
ok, i'm fine with this returning `None`; can you then please also update the docstring which is still saying "Normally this is the current master itself..."